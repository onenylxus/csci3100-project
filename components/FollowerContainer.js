// Import
import React from 'react';
import { Image, Text, View } from 'react-native';
import Style from '../assets/style';

// Export Follower container
export default function FollowerContainer({ username }) {
  return (
    <View>
      <View style={{ flexDirection: 'row', marginVertical: 20 }}>
        <Image
          style={Style.followScreenIcon}
          source={require('../assets/images/profile.png')}
        />
        <View>
          <Text
            style={{
              fontSize: 18,
              alignSelf: 'center',
              fontWeight: 'bold',
            }}
          >
            {username}
          </Text>
        </View>
      </View>
    </View>
  );
}
