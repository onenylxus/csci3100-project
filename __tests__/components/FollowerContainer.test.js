// Import
import React from 'react';
import { render } from '@testing-library/react-native';

// Import target component
import FollowerContainer from '../../components/FollowerContainer';

// Run jest tests
describe('FollowerContainer', () => {
  let element;

  beforeEach(() => {
    // Mock post
    const postMock = {
      username: 'testac',
      timestamp: new Date(2021, 4, 1),
      title: '',
      content: '',
      peopleLike: [],
      peopleDislike: [],
    };

    // Render
    element = render(<FollowerContainer post={postMock} />);
  });

  afterEach(() => {
    element.unmount();
  });

  it('matches snapshot', () => {
    expect(element.toJSON()).toMatchSnapshot();
  });
});
