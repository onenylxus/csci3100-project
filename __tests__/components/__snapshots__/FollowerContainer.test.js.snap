// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FollowerContainer matches snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginVertical": 20,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "borderRadius": 28,
          "height": 32,
          "marginHorizontal": 8,
          "width": 32,
        }
      }
    />
    <View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      />
    </View>
  </View>
</View>
`;
